### ISSUE TITLE
✅ ♿ 🐞 Bug Report: Nested Tabs use the wrong ARIA attributes

### ISSUE THREAD

**Issue by Daveden2** [elementor/elementor#22935]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

The tab widget seems to not use the proper aria attributes and roles. The code needs to be refactored.

For example, it sets a `role = "tablist"` to the group of tab contents, which I assume is for the mobile accordion, but that shouldn't be there at all. Because 
1. on mobile it is an accordion no longer a tab. 
2. a tablist can only contain children with `role="tab"`.  

Also, the `role = "tabpanel"` should be applied to the individual tab contents. 

The role `aria-selected` should only be used for the tab title while `aria-expanded`  should only be used when it switches to the mobile accordion title.

It should probably only be added when the tab switches to an accordion for mobile (and the `aria-selected` role should be removed).

There is also no focus outline set on the tab titles for keyboard users. That can be solved using custom CSS.

The correct ids are missing from the tab contents. Also, there could also be an `aria-labelledby` applied to the tabpanels.

<img width="678" alt="image" src="https://github.com/elementor/elementor/assets/105490669/9a424c74-269a-46a4-afc6-8f804374095c">

Here's the MDN docs for reference: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Tab_RoleHere are

### Steps to reproduce

1. Activate the Nested Tabs feature
2. Create a new Elementor page
3. Drop in a tab widget to the page
4. Preview the page on the frontend and test using the devtools

### Isolating the problem

- [ ] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently following the steps above.

### System Info

<details>
<summary>System Info</summary>

````txt


					
== Server Environment ==
	Operating System: Linux
	Software: LiteSpeed
	MySQL version: MariaDB Server v10.5.19
	PHP Version: 8.0.28
	PHP Memory Limit: 2048M
	PHP Max Input Vars: 5000
	PHP Max Post Size: 2048M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.2.2
	Site URL: 
	Home URL:
	WP Multisite: No
	Max Upload Size: 2 GB
	Memory limit: 40M
	Max Memory limit: 2048M
	Permalink Structure: /%postname%/
	Language: en-US
	Timezone: 0
	Admin Email: 
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor
	Version: 2.7.1
	Author: Elementor Team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36

== Active Plugins ==
	Crocoblock Wizard
		Version: 1.2.8
		Author: Crocoblock

	Dynamic.ooo - Dynamic Content for Elementor
		Version: 2.10.4
		Author: Dynamic.ooo

	Elementor
		Version: 3.14.1
		Author: Elementor.com

	Elementor Pro
		Version: 3.14.1
		Author: Elementor.com

	Hostinger
		Version: 1.4.1
		Author: Hostinger

	JetEngine
		Version: 3.2.0
		Author: Crocoblock

	JetFormBuilder
		Version: 3.0.8
		Author: Crocoblock

	JetMenu
		Version: 2.4.0
		Author: Crocoblock

	LiteSpeed Cache
		Version: 5.5
		Author: LiteSpeed Technologies

	Presto Player
		Version: 1.14.0
		Author: Presto Made, Inc

	WPCodeBox 2
		Version: 1.0.3
		Author: WPCodeBox


== Elements Usage ==
	
	wp-page : 3
		accordion : 1
		container : 16
		heading : 9
		image : 1
		loop-grid : 1
		nested-tabs : 1
		text-editor : 2
		toggle : 1
	wp-post : 0
		container : 13
		heading : 2
		icon-list : 1
		image : 2
		jet-listing-dynamic-field : 1
		loop-grid : 1
		mega-menu : 1
		text-editor : 1
	loop-item : 2
		container : 3
		heading : 3
		theme-post-excerpt : 1
		theme-post-featured-image : 2
		theme-post-title : 2


== Settings ==
	
	cpt_support: post, page, jet-menu
	disable_color_schemes: yes
	disable_typography_schemes: yes
	allow_tracking: yes
	unfiltered_files_upload: 1
	font_display: swap


== Features ==
	Custom Fonts: 0
	Custom Icons: 0

== Integrations ==
	


== Elementor Experiments ==
	Optimized DOM Output: Active
	Improved Asset Loading: Active
	Improved CSS Loading: Active
	Inline Font Icons: Inactive by default
	Accessibility Improvements: Active by default
	Additional Custom Breakpoints: Active by default
	admin_menu_rearrangement: Inactive by default
	Flexbox Container: Active
	Upgrade Swiper Library: Active
	Grid Container: Active
	Default to New Theme Builder: Active by default
	Hello Theme Header & Footer: Active
	Editor Top Bar: Inactive by default
	Landing Pages: Active by default
	Nested Elements: Active
	Nested Accordion: Inactive by default
	Lazy Load Background Images: Inactive by default
	Global Style Guide: Inactive by default
	Page Transitions: Active by default
	Notes: Active by default
	Loop: Active by default
	Form Submissions: Active by default
	Scroll Snap: Active by default
	Menu: Active


== Log ==
	
Log: showing 20 of 382023-06-21 12:54:02 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.14.0-beta5',
  'to' => '3.14.0',
)]
2023-06-21 12:54:03 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.14.0-beta4',
  'to' => '3.14.0',
)]
2023-06-21 12:54:03 [info] elementor-pro::elementor_pro_updater Started 
2023-06-21 12:54:03 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-06-21 12:54:03 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-06-21 12:54:03 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.14.0-beta4',
  'to' => '3.14.0',
)]
2023-06-21 12:54:03 [info] Elementor/Upgrades - _on_each_version Start  
2023-06-21 12:54:05 [info] Elementor/Upgrades - _on_each_version Finished 
2023-06-21 12:54:05 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.14.0-beta5',
  'to' => '3.14.0',
)]
2023-06-28 09:58:30 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.14.0',
  'to' => '3.14.1',
)]
2023-06-28 09:58:32 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.14.0',
  'to' => '3.14.1',
)]
2023-06-28 09:58:33 [info] elementor::elementor_updater Started 
2023-06-28 09:58:33 [info] Elementor/Upgrades - _on_each_version Start  
2023-06-28 09:58:33 [info] elementor-pro::elementor_pro_updater Started 
2023-06-28 09:58:33 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-06-28 09:58:33 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-06-28 09:58:33 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.14.0',
  'to' => '3.14.1',
)]
2023-06-28 09:58:34 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.14.0',
  'to' => '3.14.1',
)]
2023-06-28 09:58:36 [info] Elementor/Upgrades - _on_each_version Finished 
2023-06-28 09:58:36 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.14.0',
  'to' => '3.14.1',
)]

JS: showing 2 of 2JS: 2023-06-16 08:35:02 [error X 2][https://staging1.daveden.co.uk/wp-content/plugins/elementor/assets/js/7fada1a61d72da6d46ab.bundle.min.js:2:7805] t.target.closest(...) is null 
JS: 2023-06-29 10:54:36 [error X 2][https://staging1.daveden.co.uk/wp-content/plugins/elementor/assets/js/7fada1a61d72da6d46ab.bundle.min.js:2:7826] Cannot read properties of null (reading 'dataset') 

PHP: showing 2 of 2PHP: 2023-06-28 10:01:49 [warning X 2][/home/u496973726/domains/daveden.co.uk/public_html/staging1/wp-content/plugins/elementor/modules/history/revisions-manager.php::363] Trying to access array offset on value of type null [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-06-28 20:39:42 [warning X 2][/home/u496973726/domains/daveden.co.uk/public_html/staging1/wp-content/plugins/elementor/includes/base/controls-stack.php::1432] Undefined array key "_flex_size_tablet" [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]



== Elementor - Compatibility Tag ==
	
	Dynamic.ooo - Dynamic Content for Elementor: Compatible
	Elementor Pro: Compatible

== Elementor Pro - Compatibility Tag ==
	
	Dynamic.ooo - Dynamic Content for Elementor: Compatible
				

````

</details>

**Comment by nicholaszein** [comment]
Hi David (@Daveden2)! 

Thank you for submitting your issue, and for the detailed explanation! 🙏 

We are aware the Nested Tabs widget still needs some accessibility improvements and our team is already working on improving this and other elements like the Menu widget for better accessibility. We can’t say exactly when it'll be sorted out, but we kindly ask for your patience. 

Thanks for understanding, and thanks again for sharing your expertise with us, we really appreciate the feedback! 😉

Best regards

**Comment by Daveden2** [comment]
Glad to know it's been looked into 👍

**Comment by doronwolf** [comment]
Hi @Daveden2 I'm happy to update you that Elementor 3.16 beta 1 has a fix for that, I would really love to get your feedback on it (since it's a beta, please make sure to test it on a staging site 🙏)

**Comment by Daveden2** [comment]
Hi @doronwolf, that's great to see!
On first inspection, the tabs look okay. I'll do a bit more testing later. 

But I think the aria-label that was included, talking about "open with enter/space" may not be necessary. I believe people who use Screen readers should already know that.

**Comment by nicholaszein** [comment]
Hey @Daveden2!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved with the release of Elementor Core v3.16.0! 🥳 

Thank you so much for your contribution! 🙏 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
